src/kasse.c

```diff
@@ -186,7 +186,7 @@ static signed int buy(char *name, unsigned int price) {
      * to NULL if no such credit could be found */
     credit = find_credit(nickname);
     if (credit != NULL) {
-      while ((signed int)credit->credit < ((signed int)price * einheiten)) {
+      while ((signed int)credit->credit < (price * einheiten)) {
```

Comment:
`signed int` looks odd. Why not cast to `int32_t` instead?

Reply:
I touched as little code as possible to fix the issue. Casting to `int32_t` should be fine as well, yes.

Reply:
Let’s do that, then? :)